larray/core.py

```diff
@@ -9172,9 +9212,25 @@ def from_string(s, nb_index=None, index_col=None, sep=','):
          BE | 0 | 1
          FO | 2 | 3
     """
-    data = [[cell.strip() for cell in line.split(sep)]
-            for line in s.strip().splitlines()]
-    return from_lists(data, nb_index=nb_index, index_col=index_col)
+
+    if nb_index is not None and index_col is not None:
+        raise ValueError("cannot specify both nb_index and index_col")
+    elif nb_index is not None:
+        index_col = list(range(nb_index))
+    elif isinstance(index_col, int):
+        index_col = [index_col]
+
+    if nb_index is None and index_col is None:
+        try:
+            header = s.strip().splitlines()[0]
+            pos_last = next(i for i, v in enumerate(header.split(sep)) if
+                            isinstance(v, basestring) and '\\' in v)
+        except StopIteration:
+            pos_last = 0
+        index_col = [pos_last]
+
+    df = pd.read_csv(StringIO(s), index_col=index_col, sep=sep, skipinitialspace=True, **kwargs)
+    return df_aslarray(df)
```

Comment:
In fact, I think the whole function could be written as:

    return read_csv(StringIO(s), nb_index=nb_index, index_col=index_col, sep=sep, skipinitialspace=True, **kwargs)